copy/copy.go

```diff
@@ -861,7 +861,7 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		// re-compressed using the desired format.
 		logrus.Debugf("Blob will be converted")
 
-		compressionOperation = types.PreserveOriginal
+		compressionOperation = types.Compress
```

Comment:
I can’t see how this makes all that much sense; when the semantics of the enum can not represent all necessary operations, my first instinct would be to extend the enum to match the real world, not to double down on the semantic mismatch.

I suppose we should have separate enums for the return value of `DesiredLayerCompression` and for the `BlobInfo.CompressionOperation` values. The two are never compared directly, so we don’t _have_ to have the same for both.

Reply:
I agree that there's much space for improving the code but the issue here is that we first update the manifest and then convert it. Since we decided that Docker doesn't support zstd, we fail at updating the manifest although we ultimately want to convert.

Reply:
It's unfortunate that the Skopeo tests passed without actually checking the manifest :(

Reply:
> I agree that there's much space for improving the code but the issue here is that we first update the manifest and then convert it.

That’s just code structure we can alter.